scripts/api_diff_report/icore_module.py

```diff
@@ -0,0 +1,201 @@
+    return changed_modules
+
+
+def module_info():
+    """retrive moudle info in MODULE_LIST from `.podspecs`
+    The moudle info helps to build Jazzy
+    includes: module name, source_files, public_header_files, language, umbrella_header, framework_root
+    """
+    module_from_podspecs = module_info_from_podspecs()
+    module_list = {}
+    for k, v in module_from_podspecs.items():
+        if k in MODULE_LIST:
+            if k not in module_list:
+                module_list[k] = v
+                module_list[k]['language'] = OBJECTIVE_C if v.get(
+                    'public_header_files') else SWIFT
+                module_list[k]['scheme'] = get_scheme(k)
+                module_list[k]['umbrella_header'] = get_umbrella_header(
+                    k, v.get('public_header_files'))
+                module_list[k]['root_dir'] = get_root_dir(
+                    k, v.get('source_files'))
+
+    logging.info(f'all_module:\n{json.dumps(module_list, indent=4)}')
+    return module_list
+
+
+def get_scheme(module_name):
+    """Jazzy documentation Info SWIFT only.
+
+    Get scheme from module name in .podspecs Assume the scheme is the
+    same as the module name:
+    """
+    MODULE_SCHEME_PATCH = {
+        'FirebaseInAppMessagingSwift': 'FirebaseInAppMessagingSwift-Beta',
+    }
+    if module_name in MODULE_SCHEME_PATCH:
+        return MODULE_SCHEME_PATCH[module_name]
+    return module_name
+
+
+def get_umbrella_header(module_name, public_header_files):
+    """Jazzy documentation Info OBJC only Get umbrella_header from
+    public_header_files in .podspecs Assume the umbrella_header is with the
+    format:
+
+    {module_name}/Sources/Public/{module_name}/{module_name}.h
+    """
+    if public_header_files:
+        if isinstance(public_header_files, list):
+            return public_header_files[0].replace('*', module_name)
+        elif isinstance(public_header_files, str):
+            return public_header_files.replace('*', module_name)
+    return ''
+
+
+def get_root_dir(module_name, source_files):
+    """Get source code root_dir from source_files in .podspecs Assume the
+    root_dir is with the format:
+
+    {module_name}/Sources or {module_name}/Source
+    """
+    MODULE_ROOT_PATCH = {
+        'FirebaseFirestore': 'Firestore/Source',
+        'FirebaseFirestoreSwift': 'Firestore/Swift/Source',
+        'FirebaseCrashlytics': 'Crashlytics/Crashlytics',
+        'FirebaseInAppMessagingSwift': 'FirebaseInAppMessaging/Swift/Source',
+    }
+    if module_name in MODULE_ROOT_PATCH:
+        return MODULE_ROOT_PATCH[module_name]
+    if source_files:
+        for source_file in source_files:
+            if f'{module_name}/Sources' in source_file:
+                return f'{module_name}/Sources'
+            if f'{module_name}/Source' in source_file:
+                return f'{module_name}/Source'
+    return ''
+
+
+def module_info_from_package_swift():
```

Comment:
is this API needed. I see that it is unused.

Reply:
It was firstly used to generate module info, but replaced by `module_info_from_podspecs`.
Leave it here, in case we will use it in the future.

Reply:
I would lean towards removing it if it isn't being used with no near term plans to use it. It's fine if you'd prefer to keep it, but could you add a doc comment then to make note of its purpose and that it is unused?